src/main/ruby/core/posix.rb

```diff
@@ -161,29 +158,29 @@ def self.attach_function(method_name, native_name, argument_types, return_type =
   # ENV-related
   attach_function :getenv, [:string], :string
 
-  attach_function :setenv_native, :setenv, [:string, :string, :int], :int
+  attach_function :setenv, [:string, :string, :int], :int, as: :setenv_native
   def self.setenv(name, value, overwrite)
     Truffle.invoke_primitive :posix_invalidate_env, name
     setenv_native(name, value, overwrite)
   end
 
-  attach_function :unsetenv_native, :unsetenv, [:string], :int
+  attach_function :unsetenv, [:string], :int, as: :unsetenv_native
   def self.unsetenv(name)
     Truffle.invoke_primitive :posix_invalidate_env, name
     unsetenv_native(name)
   end
-  
+
   # Other routines
   attach_function :crypt, [:string, :string], :string
-  
+
   # Errno-related
 
   if Rubinius.linux?
-    attach_function :errno_address, :__errno_location, [], :pointer
+    attach_function :__errno_location, [], :pointer, as: :errno_address
```

Comment:
One thing I don't like with this syntax is that it's unclear which method is getting defined, I need to read until the end to know it.

Reply:
I'm not sure that criticism makes sense. The previous approach had two names listed - it was completely unclear which method is getting defined. This approach which has an `as:` label that really does make it obvious.

Reply:
The first argument is always the name of the method getting defined, the rest is meta-information for binding the native function.